### PR TITLE
Implement Mayhem

### PR DESCRIPTION
**Pull request by nottirb**
- Added a `Dockerfile` for fuzz testing
- Added `Mayhemfile(s)` for fuzz testing
- Added a `mayhem.yml` file for automatically running fuzz tests with Mayhem using the mcode action
- Wrote 3 fuzz harnesses

### CODE REVIEW

File: fuzz/fuzz_targets/rabbit.rs [hunk 1]
```diff
@@ -0,0 +1,60 @@
+use honggfuzz::fuzz;
```
Comment (xansec) [inline fuzz/fuzz_targets/rabbit.rs:1]: Asking again for confirmation of fixed random numbers dependent on test case/seed.

File: mayhem/hc-256.Mayhemfile [hunk 2]
```diff
@@ -0,0 +1,10 @@
+project: stream-ciphers
```
Comment (xansec) [inline mayhem/hc-256.Mayhemfile:1]: Please omit image name.

File: mayhem/salsa20.Mayhemfile [hunk 3]
```diff
@@ -0,0 +1,10 @@
+project: stream-ciphers
```
Comment (xansec) [inline mayhem/salsa20.Mayhemfile:1]: Please omit image name.

**Review by xansec — CHANGES_REQUESTED** [review]
Please make the requested changes.

**Comment by nottirb** [comment]
I removed the image name from the Mayhemfiles, and replaced StdRng with Pcg64 to ensure that it generates the same random number on different builds/platforms.

**Review by xansec — APPROVED** [review]
LGTM!

**Pull request merged**